Backpack/Tests/SnapshotTests/BPKCarouselSnapshotTest.swift

```diff
@@ -0,0 +1,113 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XCTest
+import Backpack
+
+import SnapshotTesting
+
+class BPKCarouselSnapshotTest: XCTestCase {
+    
+    private let imageNames = ["dialog_flare", "dialog_image", "pilanesburg-south-africa"]
+    
+    override func setUp() {
+        super.setUp()
+        isRecording = false
+    }
+    
+    func testViewSnapshotWithOneImage() {
+        let imageViews = createImageViews(count: 1)
+        let exampleView = createView(with: imageViews)
+        assertSnapshot(exampleView)
+    }
+    
+    func testViewSnapshotWithTwoImages() {
+        let imageViews = createImageViews(count: 2)
+        let exampleView = createView(with: imageViews)
+        assertSnapshot(exampleView)
+    }
+    
+    func testViewSnapshotWithThreeImages() {
+        let imageViews = createImageViews(count: 3)
+        let exampleView = createView(with: imageViews)
+        assertSnapshot(exampleView)
+    }
+    
+    func testViewSnapshotWithSecondSelected() {
+        let imageViews = createImageViews(count: 3)
+        let exampleView = createView(with: imageViews, currentImage: 1)
+        assertSnapshot(exampleView)
+    }
+    
+    func testViewSnapshotWithThirdSelected() {
+        let imageViews = createImageViews(count: 3)
+        let exampleView = createView(with: imageViews, currentImage: 2)
+        assertSnapshot(exampleView)
+    }
+}
+
+// MARK: Helpers
+extension BPKCarouselSnapshotTest {
+    private func createImageViews(count: Int) -> [UIImageView] {
+        var imageViews = [UIImageView]()
+        for index in 0..<min(count, imageNames.count) {
```

Comment:
nit-ish: 
could this not just be 0..<count? from the `createImageViews` method definition is not clear that this is happening and can be confusing while reading the code (when reading the calling code) and may lead to unexpected behaviour